example-crates/example_migrate_wallet/src/main.rs

```diff
@@ -0,0 +1,131 @@
+    let db = SqliteDatabase::new(BDK_DB_PATH);
+    let old_wallet = Wallet::new(EXTERNAL, Some(INTERNAL), NETWORK, db)?;
+
+    // Get last revealed addresses for each keychain
+    let addr = old_wallet.get_address(AddressIndex::LastUnused)?;
+    println!("Last revealed external {} {}", addr.index, addr.address);
+    let external_derivation_index = addr.index;
+    let last_revealed_external = addr.address.to_string();
+
+    let addr = old_wallet.get_internal_address(AddressIndex::LastUnused)?;
+    println!("Last revealed internal {} {}", addr.index, addr.address);
+    let internal_derivation_index = addr.index;
+    let last_revealed_internal = addr.address.to_string();
+
+    // Get unspent balance
+    let old_unspent = old_wallet.list_unspent()?;
+    let old_balance = old_wallet.get_balance()?;
+    println!("Balance before sync: {} sat", old_balance.get_total());
+
+    // Create new wallet
+    // For the new bdk wallet we pass in the same descriptors as before. If the given descriptors
+    // contain secret keys the wallet will be able to sign transactions as well.
+    let mut db = rusqlite::Connection::open(BDK_WALLET_DB_PATH)?;
+    let mut new_wallet = match bdk_wallet::Wallet::create(EXTERNAL, INTERNAL)
+        .network(bdk_wallet::bitcoin::Network::Signet)
+        .create_wallet(&mut db)
+    {
+        Ok(wallet) => wallet,
+        Err(_) => anyhow::bail!("should not have existing db"),
+    };
+
+    // Retore revealed addresses
+    let _ = new_wallet.reveal_addresses_to(
+        bdk_wallet::KeychainKind::External,
+        external_derivation_index,
+    );
+    let _ = new_wallet.reveal_addresses_to(
+        bdk_wallet::KeychainKind::Internal,
+        internal_derivation_index,
+    );
+
+    // Remember to persist the new wallet
+    new_wallet.persist(&mut db)?;
+
+    println!("\n========== New database created. ==========");
+
+    let addr = new_wallet
+        .list_unused_addresses(bdk_wallet::KeychainKind::External)
+        .last()
+        .unwrap();
+    assert_eq!(addr.to_string(), last_revealed_external);
+    println!("Last revealed external {} {}", addr.index, addr.address);
+    let addr = new_wallet
+        .list_unused_addresses(bdk_wallet::KeychainKind::Internal)
+        .last()
+        .unwrap();
+    println!("Last revealed internal {} {}", addr.index, addr.address);
+    assert_eq!(addr.to_string(), last_revealed_internal);
+
+    // Now that we migrated the wallet details, you likely want to rescan the blockchain
+    // to restore the wallet's transaction data. Here we're using the bdk_electrum client
+    // with a gap limit of 10. We also pass the `true` parameter to `full_scan` to indicate
+    // we want to collect additional transaction data (previous txouts) that are important
+    // for a fully functioning wallet.
+
+    let client = BdkElectrumClient::new(electrum_client::Client::new(ELECTRUM_URL)?);
+    let request = new_wallet.start_full_scan().inspect({
```

Comment:
Do we really need a `full_scan()` here?  Since we already know the last exposed SPK we should be able to do a `sync()` instead which will be quicker and can still retrieve the previous txouts.